File: frontend/src/services/cartService.js
import { api } from './api'

export const cartService = {
  getCart: () => api.get('/cart'),

  addToCart: (productId, quantity) => api.post('/cart/items', { productId, quantity }),

  updateCartItem: (cartItemId, quantity) => api.put(`/cart/items/${cartItemId}?quantity=${quantity}`),

  removeFromCart: (cartItemId) => api.delete(`/cart/items/${cartItemId}`),

  clearCart: () => api.delete('/cart'),
}

File: backend/src/main/java/com/ecommerce/app/service/cart/CartService.java
package com.ecommerce.app.service.cart;

import com.ecommerce.app.dto.cart.CartItemRequest;
import com.ecommerce.app.dto.cart.CartItemResponse;
import com.ecommerce.app.dto.cart.CartResponse;

public interface CartService {

    CartResponse getCartResponse(Long userId);

    CartItemResponse addToCart(Long userId, CartItemRequest request);

    CartItemResponse updateCartItemQuantity(Long userId, Long cartItemId, int quantity);

    void removeFromCart(Long userId, Long cartItemId);

    void clearCart(Long userId);
}
